### PR TITLE
[WIP] Preview components inside a directory

### PR DESCRIPTION
**Pull request by gorkalaucirica**

### CODE REVIEW

File: src/LIN3S/PatternLibraryBuilder/Symfony/Command/LoadThumbnailsCommand.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ *
+ * Copyright (c) 2017-present LIN3S <info@lin3s.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+/*
```
Comment (benatespina) [inline src/LIN3S/PatternLibraryBuilder/Symfony/Command/LoadThumbnailsCommand.php]: Remove second license header

File: src/LIN3S/PatternLibraryBuilder/Symfony/Command/LoadThumbnailsCommand.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Gorka Laucirica <gorka.lauzirika@gmail.com>
+ */
+class LoadThumbnailsCommand extends Command
+{
+    private const CHROME_BIN = "/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome";
```
Comment (benatespina) [inline src/LIN3S/PatternLibraryBuilder/Symfony/Command/LoadThumbnailsCommand.php]: This path is highly coupled with MacOS

File: src/LIN3S/PatternLibraryBuilder/Symfony/Command/LoadThumbnailsCommand.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    public function __construct(StyleguideConfigLoader $configLoader)
+    {
+        parent::__construct();
+        $this->configLoader = $configLoader;
+    }
+
+    protected function configure()
+    {
+        $this->setName('lin3s:pattern-library-builder:load-thumbnails');
```
Comment (benatespina) [inline src/LIN3S/PatternLibraryBuilder/Symfony/Command/LoadThumbnailsCommand.php]: Call setDescription method with proper explanation

File: src/LIN3S/PatternLibraryBuilder/Symfony/Command/LoadThumbnailsCommand.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $configs = $this->configLoader->allInPlain();
+
+        foreach ($configs as $config) {
+            $this->takeScreenshot($config);
+        }
+    }
+
+    private function takeScreenshot(array $config)
```
Comment (benatespina) [inline src/LIN3S/PatternLibraryBuilder/Symfony/Command/LoadThumbnailsCommand.php]: Add missing "void" return type

File: src/LIN3S/PatternLibraryBuilder/Symfony/Command/LoadThumbnailsCommand.php [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    {
+        $command = sprintf(
+            '%s --headless --disable-gpu --screenshot=%s http://localhost:8000%s?media=mobile',
```
Comment (benatespina) [inline src/LIN3S/PatternLibraryBuilder/Symfony/Command/LoadThumbnailsCommand.php]: localhost??

File: src/LIN3S/PatternLibraryBuilder/Symfony/Command/LoadThumbnailsCommand.php [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            self::CHROME_BIN,
+            __DIR__ . '/../Resources/public/thumbnails/' . mt_rand() . '.png',
```
Comment (benatespina) [inline src/LIN3S/PatternLibraryBuilder/Symfony/Command/LoadThumbnailsCommand.php]: This path is hardcoded

File: src/LIN3S/PatternLibraryBuilder/Symfony/Command/LoadThumbnailsCommand.php [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+            $config['slug'],
+            array_keys($config['config']['preview_parameters'])[0]
+        );
+
+        exec($command);
```
Comment (benatespina) [inline src/LIN3S/PatternLibraryBuilder/Symfony/Command/LoadThumbnailsCommand.php]: Maybe we can you Symfony Process to maintain better compatibility between different envs

**Comment by gorkalaucirica** [comment]
WIP = Not finished yet

**Comment by gorkalaucirica** [comment]
This feature will be implemented using another approach in #12

_Sent from my OnePlus ONEPLUS A5000 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Pull request closed without merging**